### PR TITLE
Fix tooltips and issue dependency styles

### LINKED ISSUE #13400: Tooltips go out of the screen
- Gitea version (or commit ref): 1.13.0-rc1
- Git version: 2.18.4
- Operating system: CentOS 8, bare binary
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No (need admin permissions)
- Log gist: N/A

## Description

Left side of tooltip for "May Create Git Hooks" went beyond

## Screenshots

Google Chrome 85.0:

Edit user form:
image

### PR DESCRIPTION
**Pull request by silverwind**
- Convert all CSS tooltips to JS-based ones, fixing overflow issues
- Restyle issue dependencies/dependants
- Move popup styles to base style
- CSS Helper tweaks
- Unify pseudo element selectors and lint for it

Fixes: https://github.com/go-gitea/gitea/issues/13400

Screen Shot 2020-11-07 at 16 32 09
image
Screen Shot 2020-11-07 at 17 43 01
Screen Shot 2020-11-07 at 17 43 29

### CODE REVIEW

**Comment by lafriks** [comment]
Please resolve conflicts

**Comment by silverwind** [comment]
Rebased and screenshots updated.

File: web_src/less/_repository.less [hunk 1]
```diff
@@ -1049,7 +1056,7 @@
           > .bottom.segment {
             background: #f3f4f5;
 
-            .ui.images::after {
+            .ui.images:after {
```
Comment (lafriks) [inline web_src/less/_repository.less]: This won't insert pseido element anymore...

Comment (silverwind) [reply web_src/less/_repository.less]: Should still work as browsers accepts both single and double colons, but I guess if we want to be "correct", we better convert to double colon and lint for it.

https://css-tricks.com/to-double-colon-or-not-do-double-colon/

**Comment by silverwind** [comment]
@lafriks converted to double-colon syntax and changed linting for it.

File: templates/admin/user/edit.tmpl [hunk 2]
```diff
@@ -90,7 +90,7 @@
 					</div>
 				</div>
 				<div class="inline field">
-					<div class="ui checkbox" data-tooltip="{{.i18n.Tr "admin.users.allow_git_hook_tooltip"}}">
+					<div class="ui checkbox poping up" data-content="{{.i18n.Tr "admin.users.allow_git_hook_tooltip"}}">
```
Comment (zeripath) [inline templates/admin/user/edit.tmpl]:
```suggestion
					<div class="ui checkbox poping up" data-content="{{.i18n.Tr "admin.users.allow_git_hook_tooltip"}}" data-variation="very wide">
```

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
@silverwind it's conflicting again

File: web_src/less/_repository.less (hunk 1 above)
Comment (lafriks) [reply web_src/less/_repository.less]: Both are correct but have different use cases

**Comment by lafriks** [comment]
please resolve conflict

**Comment by lafriks** [comment]
I would say that column count should not be changed because it could have undesirable effect. Generally I think single `:` is better because it does not create additional pseido elements in DOM but if we remove them it could effect spacing etc especially where we have `content: ' '` in css. So imho for now this rune should be disabled and column usage `:` or `::` should be reviewed in other PR

**Comment by silverwind** [comment]
I don't think there is any practical difference between `::before` vs `:before`. I've been using them interchangeable for a long time and I think it's purely a stylistic thing. One could argue `:` saves a few bytes and I generally prefer the single colon syntax because I don't want to think whether something is a pseudo-element or a pseudo class. Yes, I can remove these changes for now.

**Comment by lafriks** [comment]
On second thought I don't mind having `::` in this PR, so leave at it is now

**Comment by silverwind** [comment]
Rebased and squashed.

**Review by lafriks — APPROVED** [review]

**Pull request merged**